File: Java/Medium/Accepted/score-after-flipping-matrix_0.java
class Solution {
    public int matrixScore(int[][] grid) {
        
        int m = grid.length;
        int n = grid[0].length;
        for (int i = 0; i < m; i++) {
            if (grid[i][0] == 0) {
                for (int j = 0; j < n; j++) {
                    grid[i][j] = 1 - grid[i][j];
                }
            }
        }
        int sum = 0;
        for (int j = 0; j < n; j++) {
            int count = 0;
            for (int i = 0; i < m; i++) {
                if (grid[i][j] == 1) {
                    count++;
                }
            }
            sum += (Math.max(count, m - count) * (1 << (n - j - 1)));
        }
        return sum;
    }
}

File: Python3/Medium/Accepted/score-after-flipping-matrix_0.py
class Solution:
    def matrixScore(self, grid: List[List[int]]) -> int:
        
        m, n = len(grid), len(grid[0])
        for i in range(m):
            if grid[i][0] == 0:
                for j in range(n):
                    grid[i][j] ^= 1
        for j in range(1, n):
            cnt = 0
            for i in range(m):
                if grid[i][j] == 1:
                    cnt += 1
            if cnt < m - cnt:
                for i in range(m):
                    grid[i][j] ^= 1
        ans = 0
        for i in range(m):
            for j in range(n):
                ans += grid[i][j] * (1 << (n - j - 1))
        return ans